Python/pytime.c

```diff
@@ -99,9 +99,12 @@ _PyTime_Round(double x, _PyTime_round_t round)
     else if (round == _PyTime_ROUND_CEILING) {
         d = ceil(d);
     }
-    else {
+    else if (round == _PyTime_ROUND_FLOOR) {
         d = floor(d);
     }
+    else {
+        d = (d < 0.0) ? floor(d) : ceil(d);
```

Comment:
Add `assert(round == _PyTime_ROUND_UP)`.

Reply:
Done in 5c58695